### DISCUSSION TITLE
cannot update navidrome on linux since 0.54.1

### DISCUSSION THREAD

**Discussion by casi1801** [navidrome/navidrome#3594] [Q&A]
Hello,
Since two days I try to update my navidrome on Linux Ubuntu 22.04 which is installed on a Proxmox CT.
I normally do an update like this:
apt update && apt upgrade -y
su user
sudo systemctl stop navidrome.service
sudo wget https://github.com/navidrome/navidrome/releases/download/v0.54.2/navidrome_0.54.2_linux_amd64.tar.gz -O Navidrome.tar.gz
sudo tar -xvzf Navidrome.tar.gz -C /opt/navidrome/
sudo chown -R user /opt/navidrome
sudo systemctl daemon-reload
sudo systemctl start navidrome.service
sudo systemctl status navidrome.service
sudo systemctl enable navidrome.service

This worked and still works with 0.53.3 (also if I do this from 0.54.1 downwards)
If I try to update with 0.54.1 or 54.2 the navidrome.service doesn`t start

Is there anything I have to change in the structure or in the navidrome.service file ?
I cannot find any hint in the description
The installation description is still from December 2023.. so still the same..
Maybe someone can help me

The failure is:
x navidrome.service - Navidrome Music Server and Streamer compatible with Subsonic/Airsonic
     Loaded: loaded (/etc/systemd/system/navidrome.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Wed 2024-12-25 17:37:12 CET; 14s ago
    Process: 1106 ExecStart=/opt/navidrome/navidrome --configfile /var/lib/navidrome/navidrome.toml (code=exited, status=203/EXEC)
   Main PID: 1106 (code=exited, status=203/EXEC)
        CPU: 30ms

Dec 25 17:37:12 pNavidrome systemd[1]: navidrome.service: Scheduled restart job, restart counter is at 5.
Dec 25 17:37:12 pNavidrome systemd[1]: Stopped Navidrome Music Server and Streamer compatible with Subsonic/Airsonic.
Dec 25 17:37:12 pNavidrome systemd[1]: navidrome.service: Start request repeated too quickly.
Dec 25 17:37:12 pNavidrome systemd[1]: navidrome.service: Failed with result 'exit-code'.
Dec 25 17:37:12 pNavidrome systemd[1]: Failed to start Navidrome Music Server and Streamer compatible with Subsonic/Airsonic.

I already googled about that "repeated too quickly" but this only means that the service does not find something (password, link or anything else)
As remarked above... everything works with 0.53.3, installed the way explained...

Thank You for helping

**Comment by kgarner7** [comment]
The log you've shown is insufficient for seeing the original failure. You need to get additional lines to see the **actual** failure. For `systemctl status`, make sure to pass in a sufficiently large `-l` (`--lines=`) to see the original failure. Alternatively, for just the logs, `journalctl -u navidrome -n $ENOUGH_LINES_TO_SEE_THE_ERROR`

**Comment by casi1801** [comment]
user@Navidrome:/root$ **sudo systemctl status -l**
* Navidrome
    State: degraded
     Jobs: 0 queued
   Failed: 2 units
    Since: Thu 2024-12-26 08:31:02 CET; 4min 48s ago
   CGroup: /
           |-user.slice 
           | `-user-0.slice 
           |   |-session-33081.scope 
           |   | |-194 /bin/login -p --
           |   | |-584 -bash
           |   | |-609 su user
           |   | |-610 bash
           |   | |-681 sudo systemctl status -l
           |   | |-683 sudo systemctl status -l
           |   | |-684 systemctl status -l
           |   | `-685 pager
           |   `-user@0.service ...
           |     `-init.scope 
           |       |-577 /lib/systemd/systemd --user
           |       `-578 (sd-pam)
           |-init.scope 
           | `-1 /sbin/init
           `-system.slice 
             |-fail2ban.service 
             | `-211 /usr/bin/python3 /usr/bin/fail2ban-server -xf start
             |-containerd.service ...
             | `-151 /usr/bin/containerd
             |-systemd-networkd.service 
             | `-80 /lib/systemd/systemd-networkd
             |-cron.service 
             | `-186 /usr/sbin/cron -f -P
             |-docker.service ...
             | `-172 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
             |-networkd-dispatcher.service 
lines 1-35


user@Navidrome:/root$ **journalctl -u navidrome -n $ENOUGH_LINES_TO_SEE_THE_ERROR**
Hint: You are currently not seeing messages from other users and the system.
      Users in groups 'adm', 'systemd-journal' can see all messages.
      Pass -q to turn off this notice.
Dec 26 08:32:19 Navidrome navidrome[187]: time="2024-12-26T08:32:19+01:00" level=debug msg="API: Successful response" endpoint=/rest/search3.view requestId=pN>
Dec 26 08:32:19 Navidrome navidrome[187]: time="2024-12-26T08:32:19+01:00" level=debug msg="HTTP: POST http://192.178.168.2:4711/rest/search3.view" elapsedTi>
Dec 26 08:32:19 Navidrome navidrome[187]: time="2024-12-26T08:32:19+01:00" level=debug msg="API: New request /rest/getSong.view" client="Music Assistant" requ>
Dec 26 08:32:19 Navidrome navidrome[187]: time="2024-12-26T08:32:19+01:00" level=debug msg="Found matching player" client="Music Assistant" id=138f81af-0f64-4>
Dec 26 08:32:19 Navidrome navidrome[187]: time="2024-12-26T08:32:19+01:00" level=debug msg="API: Successful response" endpoint=/rest/getSong.view requestId=pN>
Dec 26 08:32:19 Navidrome navidrome[187]: time="2024-12-26T08:32:19+01:00" level=debug msg="HTTP: POST http://192.178.168.2:4711/rest/getSong.view" elapsedTi>
Dec 26 08:32:39 Navidrome navidrome[187]: time="2024-12-26T08:32:39+01:00" level=info msg="Stopping HTTP server"
Dec 26 08:32:39 Navidrome navidrome[187]: time="2024-12-26T08:32:39+01:00" level=debug msg="Scheduler: stop"
Dec 26 08:32:42 Navidrome navidrome[187]: time="2024-12-26T08:32:42+01:00" level=info msg="Closing Database"
Dec 26 08:32:42 Navidrome navidrome[187]: time="2024-12-26T08:32:42+01:00" level=info msg="Navidrome stopped, bye."

another try after reboot the disk:
user@Navidrome:/root$ **sudo journalctl -u navidrome -n $ENOUGH_LINES_TO_SEE_THE_ERROR**
Dec 26 08:50:56 Navidrome systemd[1]: Started Navidrome Music Server and Streamer compatible with Subsonic/Airsonic.
Dec 26 08:50:56 Navidrome systemd[754]: navidrome.service: **Failed to locate executable /opt/navidrome/navidrome: Permission denied**
Dec 26 08:50:56 Navidrome systemd[754]: navidrome.service: Failed at step EXEC spawning /opt/navidrome/navidrome: Permission denied
Dec 26 08:50:56 Navidrome systemd[1]: navidrome.service: Main process exited, code=exited, status=203/EXEC
Dec 26 08:50:56 Navidrome systemd[1]: navidrome.service: Failed with result 'exit-code'.
Dec 26 08:50:56 Navidrome systemd[1]: navidrome.service: Scheduled restart job, restart counter is at 5.
Dec 26 08:50:56 Navidrome systemd[1]: Stopped Navidrome Music Server and Streamer compatible with Subsonic/Airsonic.
Dec 26 08:50:56 Navidrome systemd[1]: navidrome.service: Start request repeated too quickly.
Dec 26 08:50:56 Navidrome systemd[1]: navidrome.service: Failed with result 'exit-code'.
Dec 26 08:50:56 Navidrome systemd[1]: Failed to start Navidrome Music Server and Streamer compatible with Subsonic/Airsonic.

so what is the difference to the executable file in 0.53.3 to the 0.54.x??

**Comment by TheWrathKing** [comment]
Hello @casi1801 ,
Seems like a permissions problem
Your log indicates : 
Dec 26 08:50:56 Navidrome systemd[754]: navidrome.service: Failed to locate executable /opt/navidrome/navidrome: Permission denied
Dec 26 08:50:56 Navidrome systemd[754]: navidrome.service: Failed at step EXEC spawning /opt/navidrome/navidrome: Permission denied

Maybe try a chmod +x

**Reply by casi1801** [reply]
Hello @TheWrathKing,
thank you very much! That worked.
I did
sudo chmod +x /opt/navidrome/navidrome
after that the navidrome.service started!
